src/components/Launch/index.js

```diff
@@ -29,38 +29,53 @@ export class Launch extends React.Component {
         const data = this.props.data;
         return (
             <section className="launch" >
-                <LaunchImage links={data.links} imgBool={this.state.imgBool} />
+                <Route path={"/" + data.mission_name}
+                    children={({ match }) => {
+                        if (match !== null) {
+                            return (
+                                <LaunchImage links={data.links} isShowingImages={true} isCurrentRoute={true}/>
+                            );
+                        }
+                        else{
+                            return (
+                                <LaunchImage links={data.links} isShowingImages={false} isCurrentRoute={false}/>
+                            );
+                        }
+                    }} />
                 <h2 className="launchTitle">{data.mission_name}</h2>
                 <button className="favoriteButton"
                     onClick={(e) => {//dispatch favorites actions here.
-                        if (!this.props.heart) {
+                        if (!this.props.isFavorited) {
```

Comment:
This `onClick` should be pulled out into a class method.

It's _always_ a best practice to _not_ inline functions within a render. The reason is because React "re-renders" very often and that means that each time the component re-renders you are creating a new function every time.

If you pull this out into a class method, React will "cache" the instance of this method and it will save you on some function creation/rendering.